lib/crawly/utils.ex

```diff
@@ -113,4 +113,50 @@ defmodule Crawly.Utils do
   def send_after(pid, message, timeout) do
     Process.send_after(pid, message, timeout)
   end
+
+  @doc """
+  A helper which allows to extract a given setting.
+
+  Returned value is a result of intersection of the global settings and settings
+  defined as settings_override inside the spider. Settings defined on spider are
+  taking precedence over the global settings defined in the config.
+  """
+  @spec get_settings(setting_name, spider_name, default) :: result
+    when setting_name: atom(),
+         spider_name: atom(),
+         default: term(),
+         result: term()
+
+  def get_settings(setting_name, spider_name \\ nil, default \\ nil) do
+    global_setting = Application.get_env(:crawly, setting_name, default)
+    case get_spider_setting(setting_name, spider_name) do
```

Comment:
For example:
```
def get_settings(setting_name, spider_name \\ nil , defualt \\ nil) do
    global = Application.get_env(:crawly, setting_name, default)
    overrides = cond do
        function_exported?(spider_name, :override_settings, 1) when is_atom(spider_name) ->
            spider_name.override_settings(global)
        function_exported?(spider_name, :override_settings, 0) when is_atom(spider_name) ->
            spider_name.override_settings()
        true -> 
           []
    end
    merged = Keyword.merge(global, overrides)
    Keyword.get(merged, setting_name, default)
end
```

Allows for more advanced logic in the override function, and the user doesn't have to do a `Application.get_env` in the spider module.

Reply:
I am not quite sure about it. Lets maybe discuss it in a separate issue.